BUILD.bazel

```diff
@@ -0,0 +1,209 @@
+        "OgreMain/include/Compositor/**/*.h",
+        "OgreMain/include/CommandBuffer/*.h",
+        "OgreMain/include/Math/Array/*.h",
+        "OgreMain/include/Math/Simple/*.h",
+        "OgreMain/include/Math/Simple/C/*.h",
+        "OgreMain/include/Math/Simple/C/*.inl",
+        "OgreMain/include/Math/Array/SSE2/Single/*.h",
+        "OgreMain/include/Math/Array/SSE2/Single/*.inl",
+    ],
+) + [
+    "OgreMain/src/OgreImageResampler.h",
+    "OgreMain/src/OgrePixelConversions.h",
+    "OgreMain/src/OgreSIMDHelper.h",
+]
+
+cc_library(
+    name = "ogre_main_headers",
+    hdrs = main_hdrs + ["OgreMain/include/OgreBuildSettings.h"],
+    includes = [
+        "OgreMain/include",
+        "OgreMain/include/Animation",
+        "OgreMain/include/CommandBuffer",
+        "OgreMain/include/Compositor",
+        "OgreMain/include/GLX",
+        "OgreMain/include/Hash",
+        "OgreMain/include/Math",
+        "OgreMain/include/Threading",
+        "OgreMain/include/stbi",
+    ],
+)
+
+ogre_gl_support_hdrs = glob(
+    [
+        "RenderSystems/GL3Plus/include/*.h",
+        "RenderSystems/GL3Plus/include/GL/*.h",
+        "RenderSystems/GL3Plus/include/GLSL/*.h",
+        "RenderSystems/GL3Plus/include/Vao/*.h",
+        "RenderSystems/GL3Plus/include/windowing/*.h",
+        "RenderSystems/GL3Plus/include/windowing/GLX/*.h",
+        "RenderSystems/GL3Plus/src/windowing/*.h",
+        "RenderSystems/GL3Plus/src/windowing/GLX/*.h",
+    ],
+)
+
+ogre_gl_support_includes = [
+    "RenderSystems/GL3Plus/include",
+    "RenderSystems/GL3Plus/include/GL",
+    "RenderSystems/GL3Plus/include/GLSL",
+    "RenderSystems/GL3Plus/include/Vao",
+    "RenderSystems/GL3Plus/include/windowing",
+    "RenderSystems/GL3Plus/include/windowing/GLX",
+    "RenderSystems/GL3Plus/src/GLSL/include",
+    "RenderSystems/GL3Plus/src/windowing",
+    "RenderSystems/GL3Plus/src/windowing/GLX",
+    "RenderSystems/GL3Plus/include",
+    "RenderSystems/GL3Plus/include/GLSL",
+    "RenderSystems/GL3Plus/src/windowing/GLX",
+    "RenderSystems/GL3Plus/include/windowing/GLX",
+]
+
+ogre_gl_support_srcs = glob(
+    [
+        "RenderSystems/GL3Plus/src/windowing/GLX/*.cpp",
+        "RenderSystems/GL3Plus/src/GLSL/*.cpp",
+        "RenderSystems/GL3Plus/src/Vao/*.cpp",
+        "RenderSystems/GL3Plus/src/*.cpp",
+    ],
+)
+
+cc_library(
+    name = "ogre_main",
+    srcs = main_srcs + ogre_gl_support_srcs + ogre_gl_support_hdrs,
+    hdrs = main_hdrs + ogre_gl_support_hdrs,
+    copts = [
+        "-Wno-implicit-fallthrough",
+        "-Wno-string-conversion",
+        "-Wno-deprecated-declarations",
+    ],
+    defines = ["OGRE_STATIC_LIB=1"],
+    includes = ogre_gl_support_includes,
+    linkopts = [
+        "-ldl",
+        "-pthread",
+    ],
+    deps = [
+        ":ogre_main_headers",
+        "@X//:SM",
+        "@X//:Xaw",
+        "@X//:Xrandr",
+        "@freeimage",
+        "@gl//:GL",
+        "@gl//:GLU",
+        "@gl//:glut",
```

Comment:
do we actually use `glut`?

Reply:
We had it in the CMake, but it appears to work fine without it, so I have removed it.